src/appengine/handlers/crash_query.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Handler for the crash query api."""
+
+from flask import request
+
+from crash_analysis import crash_analyzer
+from crash_analysis.stack_parsing import stack_analyzer
+from datastore import data_handler
+from handlers import base_handler
+from libs import auth
+from libs import handler
+from libs import helpers
+
+
+class Handler(base_handler.Handler):
+  """Handler that gets the crash stats when user first lands on the page."""
+
+  @handler.post(handler.JSON, handler.JSON)
+  @handler.oauth
+  def post(self):
+    """Handle a post request."""
+    if not auth.get_current_user():
```

Comment:
Does handler.oauth decorator not check this ?

Reply:
Nope. Was thinking to change this, but would prefer to do this in a separate PR in case things break.

Reply:
Looking at this again. We can't change handler.oauth to enforce this, as it's used on handlers for web stuff too (e.g. the testcase refresh handler). It was intended to add an additional way for authenticating against an endpoint, but does not make it exclusive.

Reply:
Thanks for checking, yes it looks like it just adds additional headers but does not validate them.